SupportFiles/ws-tools.sh

```diff
@@ -0,0 +1,161 @@
+#!/bin/bash
+
+# The "StackName" will be replaced with the AWS::StackName during runtime
+function err_exit {
+	echo "${1}"
+	logger -p kern.crit -t ws-tools.sh "${1}"
+	/opt/aws/bin/cfn-signal -e 1 --stack StackName --resource Ec2instance
+	exit 1
+}
+
+# Install GNOME
+yum -y groups install 'GNOME Desktop' || err_exit "Failed to install GNOME Desktop."
+systemctl set-default graphical.target
+
+# Install VNC server
+yum -y install tigervnc-server || err_exit "Failed to install TigerVNC Server."
+
+# Generate default VNC server password
+# The "VNCServerPaswd" will be replaced with the VNCServerPasswd parameter and "WorkstationUser" with the WorkstationUser parameter in the CFN during runtime
+umask 0077
+mkdir -p /home/WorkstationUser/.vnc
+chmod go-rwx /home/WorkstationUser/.vnc
+vncpasswd -f <<<VNCServerPasswd> /home/WorkstationUser/.vnc/passwd
+chown -R WorkstationUser:WorkstationUser /home/WorkstationUser/.vnc
+
+# Configure VNC server
+cp /lib/systemd/system/vncserver@.service  /etc/systemd/system/vncserver@:1.service
+sed -i 's/<USER>/WorkstationUser/g' /etc/systemd/system/vncserver@:1.service
+systemctl daemon-reload
+systemctl start vncserver@:1
+systemctl enable vncserver@:1
+
+# Add firewall VNC server firewall rules
+setenforce 0
+firewall-cmd --add-port=5901/tcp
+firewall-cmd --add-port=5901/tcp --permanent
+setenforce 1
+
+# Unzip the tools.tar.gz
+tar xfz /etc/cfn/tools/tools.tar.gz -C /etc/cfn/tools || err_exit "Failed to unzip tools.tar.gz"
+
+# Install Anaconda
+bash /etc/cfn/tools/anaconda/anaconda.sh -b -p /home/WorkstationUser/anaconda3
+chown -R WorkstationUser:WorkstationUser /home/WorkstationUser/anaconda3
+
+cp /etc/cfn/tools/anaconda/anaconda.desktop /usr/share/applications/anaconda.desktop
+cp /etc/cfn/tools/anaconda/anaconda.desktop /home/WorkstationUser/.local/share/applications/anaconda.desktop
+chown WorkstationUser:WorkstationUser /home/WorkstationUser/.local/share/applications/anaconda.desktop
+chmod 600 /home/WorkstationUser/.local/share/applications/anaconda.desktop
```

Comment:
The owner was root for /home/WorkstationUser/anaconda3. Need to change to the workstation owner; otherwise, the app won't be displayed in the application menu on the GUI.